spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/convert/QueryMapper.java

```diff
@@ -1170,7 +1170,7 @@ public String getMappedKey() {
 
 			if (sourceProperty != null && sourceProperty.getOwner().equals(entity)) {
 				return mappingContext
-						.getPersistentPropertyPath(PropertyPath.from(sourceProperty.getName(), entity.getTypeInformation()));
+						.getPersistentPropertyPath(PropertyPath.from(Pattern.quote(sourceProperty.getName()), entity.getTypeInformation()));
```

Comment:
Wouldn't it rather make sense to fix that issue in Commons? Property path creation does all sorts of string transformations and lookups (Decapitalize, `owningType.getProperty(propertyName)`) while regex matching is a small part of the entire process.

Reply:
it is yes - for the upcoming service release I'd tough rather have the regression fixed in this project and investigate a fix for commons in the upcoming release cycle as it may lead to unintentional behaviour in other modules.